blocktx/async/nats_mq/consumer.go

```diff
@@ -0,0 +1,74 @@
+package nats_mq
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+const (
+	consumerQueue = "register-tx-group"
+	topic         = "register-tx"
+)
+
+type Consumer struct {
+	logger       *slog.Logger
+	nc           *nats.Conn
+	topic        string
+	txChannel    chan []byte
+	subscription *nats.Subscription
+}
+
+func NewNatsMQConsumer(txChannel chan []byte, logger *slog.Logger, natsURL string) (*Consumer, error) {
+
+	var nc *nats.Conn
+
+	var err error
+	for i := 0; i < 5; i++ {
```

Comment:
Are we trying five times for it to connect?

Reply:
Yes, this tries 5 times and once that has worked it exits. But I think it should return an error if it didn't work after 5 times. Need to add that.